pkg/trace/fileserver.go

```diff
@@ -263,7 +263,9 @@ func (lt *LocalTracer) PushAll() error {
 
 // S3Download downloads files that match some prefix from an S3 bucket to a
 // local directory dst.
-func S3Download(dst, prefix string, cfg S3Config) error {
+// fileNames is a list of traced jsonl file names to download. If it is empty, all traces are downloaded.
+// fileNames should not have .jsonl suffix.
+func S3Download(dst, prefix string, fileNames []string, cfg S3Config) error {
```

Comment:
this is technically a Go API breaking change and pkg/trace is public

Reply:
That is a good point, I'll defer it to @evan-forbes, do you think we should avoid merging this change? I can also re-implement it in a backward compatible way i.e., by introducing two different S3Download methods, with and without fileNames.

Reply:
nice let's do all things backwards compatible if we can out of good habit. If there significant cost to doing them backwards compatible then we can reevaluate.